compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4732,30 +4732,19 @@ static TR::Register * inlineSinglePrecisionSQRT(TR::Node *node, TR::CodeGenerato
   return node->getRegister();
 }
 
-TR::Register* OMR::X86::TreeEvaluator::performSimpleAtomicMemoryUpdate(TR::Node* node, int8_t size, TR_X86OpCodes op, TR::CodeGenerator* cg)
+TR::Register* OMR::X86::TreeEvaluator::performSimpleAtomicMemoryUpdate(TR::Node* node, TR_X86OpCodes op, TR::CodeGenerator* cg)
    {
-   // There are two versions of native exchange helpers:
-   //     2 nodes: exchange with [address]
-   //     3 nodes: exchange with [address+offset]
-   TR::Register* object = cg->evaluate(node->getFirstChild());
-   TR::Register* offset = node->getNumChildren() == 2 ? NULL : cg->evaluate(node->getSecondChild());
-   TR::Register* value = TR::TreeEvaluator::intOrLongClobberEvaluate(node->getLastChild(), size == 8, cg);
-   // Assume that the offset is positive and not pathologically large (i.e., > 2^31).
-   if (offset && TR::Compiler->target.is32Bit() && size == 8)
-      {
-      offset = offset->getLowOrder();
-      }
+   TR::Register* address = cg->evaluate(node->getChild(0));
+   TR::Register* value   = cg->evaluate(node->getChild(1));
+   TR::Register* result  = cg->allocateRegister();
 
-   generateMemRegInstruction(op, node, generateX86MemoryReference(object, offset, 0, cg), value, cg);
+   generateRegRegInstruction(MOVRegReg(), node, result, value, cg);
```

Comment:
That may be true, but there's nothing in the way of documentation or an assert to enforce this.  Because the original code at least *attempted* (or at least *appeared to attempt*) to support 32-bit targets and 8-byte updates and you deleted it as part of this PR, I'd say its fair to replace it with an assert at the entry of this function if some code attempts to call it under those conditions.

Reply:
Good point. I've added an assertion at the entry so that 8-byte updates on IA32 should be detected; and it also clarifies that this method does not intend to support such operation.